src/components/drag-and-drop/drag-and-drop.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { UploadIcon } from '../../svgs';
+import { DragAndDropProps } from './types';
+import { ComponentClassNames } from '../../utils/constants';
+import { getClassNameByModifier } from '../../utils/classname-modifiers';
+
+export const DragAndDrop = React.forwardRef<HTMLDivElement, DragAndDropProps>(
+  (
+    {
+      title,
+      description,
+      dragging,
+      isHidden,
+      setDragging,
+      handleDroppedFiles,
+      backgroundVariant,
+      className,
+      children,
+      ...rest
+    },
+    ref
+  ) => {
+    const componentClasses = classNames(
+      ComponentClassNames.DragAndDrop,
+      className
+    );
+    const componentOverlayClasses = classNames(
+      getClassNameByModifier(ComponentClassNames.DragAndDrop, 'overlay'),
+      getClassNameByModifier(ComponentClassNames.DragAndDrop, backgroundVariant)
+    );
+
+    let dragCounter = 0;
```

Comment:
can we handle dragCounter in the state or it's hard to manage?

Reply:
@dhruvilsaleshandy `dragCounter` is just a global variable that we need to keep the count, whether the user is still dragging a file or not. Making it a state variable is not a good idea as it will update when the user is dragging the file, so at that time we don't need to mutate any state, and it will also prevent unnecessary re-renders